src/main/antlr4/com/yahoo/validatar/assertion/Grammar.g4

```diff
@@ -33,51 +38,91 @@ import java.util.HashMap;
         return lookedUpValues;
     }
 
-    private String getColumnValue(String name) {
-        String result = row.get(name);
-        lookedUpValues.put(name, result);
+    private TypedObject getColumnValue(String name) {
+        TypedObject result = row.get(name);
+        if (result == null) {
+            throw new RuntimeException("Unable to find value for column: " + name + " in results");
+        }
+        lookedUpValues.put(name, result.data.toString());
         return result;
     }
 
     private String stripQuotes(String literal) {
         return literal.substring(1, literal.length() - 1);
     }
+
+    private TypedObject parseString(String text) {
+        return new TypedObject(StringEscapeUtils.unescapeJava(stripQuotes(text)), Type.STRING);
+    }
+
+    private TypedObject parseWholeNumber(String text) {
+        TypedObject object;
+        try {
+            object = new TypedObject(Long.parseLong(text), Type.LONG);
+        } catch (NumberFormatException nfe) {
+            object = new TypedObject(new BigDecimal(text), Type.DECIMAL);
```

Comment:
If there is a literal like 123a4 > 123 in the assertion, this will throw an exception which will cause termination of the tests. I raised an enhancement to fix the underlying issue here https://github.com/yahoo/validatar/issues/6